Fix a misbehaviour of -L and some code formatting tests.

shlomif opened the pull request:
Hi!

This pull-req fixes the code formatting tests with clang-format (I'm using clang-3.8.0-9.mga6 on mageia x86-64 v6) and in a separate commit - adds a test and a fix to a bug in -L.

Please consider applying.

-- @shlomif

ggreer commented:
Nice! I only have one suggestion: Can you please remove the comment above line 182 in search.c? Since you fixed the bug, it's no longer relevant.

shlomif commented:
@ggreer - thanks for the input! Done in https://github.com/ggreer/the_silver_searcher/pull/888/commits/a75e16fbe2d3f0301daa10c8fde4e9e4d505e243

shlomif commented:
Hi! Any news? Can this pull-req be merged?

shlomif commented:
Hi! Any news? Can this pull-req be merged? Ping.

shlomif commented:
Dear @ggreer  , please let me know if this project is still active or if you need any help.

shlomif commented:
OK, fixed the build and the tests and rebased. Nevertheless, this pull-req should have been merged a long time ago and this repository accumulated 74 others. When will they get merged? In the meanwhile, I see there have been a few 1.0.x releases...

shlomif commented:
I ran into this very bug today again. Please merge it already.
